base/threaded_list.hpp

```diff
@@ -12,9 +12,28 @@ class ThreadedList : public ThreadedContainer
 private:
 
   std::list<T> m_list;
-  bool m_isEmpty;
-  std::string m_resName;
+  std::atomic<bool> m_isEmpty;
 
+  bool WaitNonEmpty(std::unique_lock<std::mutex> &lock)
+  {
+    bool doFirstWait = true;
+
+    while ((m_isEmpty = m_list.empty()))
+    {
+      if (IsCancelled())
+        break;
+
+      if (doFirstWait)
+        doFirstWait = false;
```

Comment:
What is it used for?

Reply:
Have no idea. May have been used before for debugging, but has not been removed. 
I removed "extra" code from this function.

Reply:
If you already touched the code, then make it better ;-)